### PR TITLE
feat(server): pick free port if none specified

### PR DESCRIPTION
**Pull request by marvinhagemeister**
Noticed that it's a little annoying running multiple fresh projects at the same time because of trying to use the same port by default. When running locally during development I don't really care what port it's on, so we can just pick a free one ourselves. The logic is the same as found in vite or wmr.

This should also allow us to run tests in parallel.

### CODE REVIEW

**Comment by marvinhagemeister** [comment]
We'll definitely want to enable parallel testing at some point. It's much faster. Will PR that later today.

| Mode | Time |
|---|--:|
| Sequential | 43s |
| Parallel | 12s |

**Review by lucacasonato — COMMENTED** [review]
Can we compute the ctx.handler() once instead of multiple times?

**Comment by marvinhagemeister** [comment]
@lucacasonato good idea! Pushed a commit that hoists the handler creation 👍

File: src/server/mod.ts [hunk 1]
```diff
@@ -86,10 +85,51 @@ export async function start(routes: Manifest, opts: StartOptions = {}) {
...
+    // That way the user only needs to increment a number when running
+    // multiple apps vs having to remember completely different ports.
+    let firstError;
+    for (let port = 8000; port < 8020; port++) {
+      try {
+        await bootServer(handler, { ...opts, port });
+        firstError = undefined;
+        break;
+      } catch (err) {
+        if (err.code === "EADDRINUSE") {
```
Comment (lucacasonato) [inline src/server/mod.ts]: This is correct - but `err instanceof Deno.errors.AddrInUse` is more Deno idiomatic.

**Review by lucacasonato — APPROVED** [review]

File: src/server/mod.ts (hunk 1 above)
Comment (marvinhagemeister) [reply src/server/mod.ts]: Good point, I'll change that.

**Pull request merged**